xmppserver/src/main/java/org/jivesoftware/util/ListPager.java

```diff
@@ -333,6 +373,18 @@ public String getPageFunctions() {
             .append("\t}\n")
             .append("\n");
 
+        sb.append("\tfunction toggleColumnOrder(sortColumnNumber) {\n")
+            // Orders based on a particular column, reversing the order if that column was already ordered on.
+            .append(String.format("\t\tvar formObject = document.getElementById('%s');\n", PAGINATION_FORM_ID))
+            .append(String.format("\t\tvar previousSortColumnNumber = formObject.%s.value || '0';\n", "sortColumnNumber"))
```

Comment:
These next few lines seem to use `String.format` unnecessarily - passing in a string constant that may as well just have been in the format string? should `"sortColumnNumber"` have been a `private static final String` definition and used elsewhere? Possibly `REQUEST_PARAMETER_KEY_SORT_COLUMN_NUMBER `(and similarly for the next few lines)

Reply:
Usage of `String.format` with constants predates my involvement here :) and to be honest, I like how it improves readability of the text in which the value is injected. The escaped javascript is difficult enough to read in itself. Doing more string concatting would not improve readability, I think.

Reply:
I definitely intended those strings to be replaced by the constants that were already defined. I'll push that change.